ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/commons/costevaluatorresult/AbstractStateWiseCostEvaluatorResult.java

```diff
@@ -29,28 +29,23 @@ protected AbstractStateWiseCostEvaluatorResult(Map<State, Double> costPerState,
         this.costlyElements = costlyElements;
     }
 
-    private DoubleStream filterCostsOnContingency(Set<String> contingenciesToExclude) {
-        return costPerState.entrySet().stream().filter(entry -> filterStateOnContingency(entry.getKey(), contingenciesToExclude)).mapToDouble(Map.Entry::getValue);
-    }
-
     @Override
     public double getCost(Set<String> contingenciesToExclude) {
-        return evaluateResultsWithSpecificStrategy(filterCostsOnContingency(contingenciesToExclude));
+        DoubleStream filteredCosts = costPerState.entrySet().stream()
+            .filter(entry -> stateHasContingencyToExclude(entry.getKey(), contingenciesToExclude))
+            .mapToDouble(Map.Entry::getValue);
+        return evaluateResultsWithSpecificStrategy(filteredCosts);
     }
 
     protected abstract double evaluateResultsWithSpecificStrategy(DoubleStream filteredCostsStream);
 
     @Override
     public List<FlowCnec> getCostlyElements(Set<String> contingenciesToExclude) {
-        return costlyElements.stream().filter(flowCnec -> filterFlowCnecOnContingency(flowCnec, contingenciesToExclude)).toList();
+        return costlyElements.stream().filter(flowCnec -> stateHasContingencyToExclude(flowCnec.getState(), contingenciesToExclude)).toList();
     }
 
-    private static boolean filterStateOnContingency(State state, Set<String> contingenciesToExclude) {
+    private static boolean stateHasContingencyToExclude(State state, Set<String> contingenciesToExclude) {
         Optional<Contingency> contingency = state.getContingency();
         return contingency.isEmpty() || !contingenciesToExclude.contains(contingency.get().getId());
     }
```

Comment:
The name doesnt correspond to what is returned. I would change the return value rather than the naming to avoid double negatives.

Reply:
I replaced the return value by `contingency.isPresent() && contingenciesToExclude.contains(contingency.get().getId())` but Sonar did not like that the value was always negated. So I went for `statesContingencyMustBeKept` instead.